### DISCUSSION TITLE
DagsterTypes cannot be used to annotate a config type

### DISCUSSION THREAD

**Discussion by yuval-airis** [dagster-io/dagster#18360] [Q&A] [answered]
I got an unclear error running my system, and I was unsure what part of the code triggered it:
`Could not load location main.py to check for sensors due to the following error: dagster._core.errors.DagsterInvalidDagsterTypeInPythonicConfigDefinitionError: Error defining Dagster config class 'MachineForwardingConfig' on field 'file_keys'. DagsterTypes cannot be used to annotate a config type. DagsterType is meant only for type checking and coercion in op and asset inputs and outputs.`

I defined this:

class MachineForwardingConfig(Config):
    file_keys: List[str]

I thought that perhaps I needed to use Dagster's String class instead, but I got another error when I did:

`dagster._core.errors.DagsterInvalidConfigDefinitionError: Error defining config. Original value passed: <class 'inspect._empty'>. <class 'inspect._empty'> cannot be resolved.`

The call itself was:

>     materialize(
>         [machine_forwarding],
>         run_config=RunConfig(
>             ops={ "machine_forwarding": MachineForwardingConfig(file_keys)}
>         )
>     )

(machine_forwarding being the name of function defined as an asset, and therefore an op)

The op itself had no other params:

`@op

def machine_forwarding(config):`

Is that an incorrect way to call that op?

**Comment by smackesey** [comment, marked as answer]
Hi @yuval-airis, if you import `List` from `typing` instead of `dagster` it should work. The name collisions of top-level exports from Dagster with typing are a relic of Python 2 support and we should probably drop them.
